src/HDWallet.cpp

```diff
@@ -62,12 +70,18 @@ HDWallet::~HDWallet() {
     std::fill(passphrase.begin(), passphrase.end(), 0);
 }
 
-void HDWallet::updateEntropy() {
-    // generate entropy (from mnemonic)
-    Data entropyRaw(32 + 1);
-    auto entropyBits = mnemonic_to_bits(mnemonic.c_str(), entropyRaw.data());
+void HDWallet::updateSeedAndEntropy() {
+    // generate seed from mnemonic
+    // it is assumed that mnemonic is valid, enforced before calling
+    mnemonic_to_seed(mnemonic.c_str(), passphrase.c_str(), seed.data(), nullptr);
+
+    // generate entropy from mnemonic
+    Data entropyRaw((Mnemonic::MaxWords * 11) / 8);
```

Comment:
what are these magic numbers?

Reply:
11 is BitsPerWord, added now as constant.  8 is bits/byte.